src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useMemo } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, PositionRotation } from "../../constants/interfaces";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+const MetadataLabels: Record<keyof AgentMetadata, JSX.Element> = {
+    uniqueId: <span>Unique ID</span>,
+    agentType: <span>Agent Type</span>,
+    position: <span>Position</span>,
+    rotation: (
+        <>
+            Rotation
+            <br />
+            (degrees)
+        </>
+    ),
+    radius: <span>Radius</span>,
+};
+
+const formatFloatForDisplay = (float: number): string => {
+    return parseFloat(float.toPrecision(2)).toString();
```

Comment:
I think it would be good to put this in utils, and format these numbers when you save the data instead of saving them in state in a format you'll never use